android-sdk/src/main/java/ai/evolv/android_sdk/EvolvStoreImpl.java

```diff
@@ -697,26 +737,82 @@ private void evaluateBranch(JsonElement context,
         return expLoadedList;
     }
 
-    // TODO: 07.07.2021 need to test+
+    //need for testing (unit test)
+    public void setGenomes(JsonObject genomes) {
+        this.genomes = genomes;
+    }
+
+    //need for testing (unit test)
+    public void setActiveKeys(JsonObject activeKeys) {
+        this.activeKeys = activeKeys;
+    }
+    //todo uncomment (callBack testing)
+//    JsonObject getActiveKeys(String prefix) {
+//        JsonObject result = new JsonObject();
+//
+//        for (Map.Entry<String, JsonElement> key : activeKeys.entrySet()) {
+//            if (hasPrefix(key.getValue().getAsString(), prefix)) {
+//                result.addProperty("current_" + key.getKey(), key.getValue().getAsString());
+//            }
+//        }
+//        return result;
+//    }
+
+    void subscribe(EvolvType type, String value, EvolvCallBack callBack) {
+        Pair<String, EvolvType> pair = new Pair<>(value, type);
+        subscriptions.put(callBack, pair);
+
+        performAction(type, value, callBack);
+    }
+
+    private void performAction(EvolvType type, String value, EvolvCallBack callBack) {
+        switch (type) {
+            case getActiveKeys: {
+                if (value.isEmpty()) {
+                    JsonObject activeKeys = getActiveKeys();
+                    callBack.invoke(activeKeys);
+                } else {
+                    JsonObject activeKeysPrefix = getActiveKeys(value);
+                    callBack.invoke(activeKeysPrefix);
+                }
+                break;
+            }
+            case get: {
+                JsonElement element = getValue(value);
+                JsonElement result = JsonNull.INSTANCE;
+
+                if (element == null) {
+                    result = JsonNull.INSTANCE;
```

Comment:
Result is already JsonNull.INSTANCE

Reply:
when we want to get some value by key, this value may not be there. and then we need to say that there is no such value, we can send a message in Jason "there is no such value" or send a "null" value. For now, I have chosen the second case

Reply:
if we need something else I will redo